jetty-core/jetty-io/src/main/java/org/eclipse/jetty/io/ByteBufferPool.java

```diff
@@ -92,6 +93,40 @@ public void clear()
         }
     }
 
+    /**
+     * A ByteBufferPool for a specific size and type of buffer
+     */
+    class Sized extends Wrapper
```

Comment:
+1 to this, we do pass the (ByteBufferPool, direction, size) triplet too often and this is a nice helper.

Reply:
I still think the introduction of this class is orthogonal to this PR, that it should not be an _is-a_ BBP but just delegate to BBP, and it causes behavior changes that may be unexpected in minor Jetty releases.
I'd rather introduce it in 12.1.x.

Reply:
This class follows the pattern of the ThreadPool.SizedThreadPool.  It is just a normal BBP with an extra no-args method to get a specifically sized pool.     The name also follows the pattern from ThreadPool.    This doesn't introduce any changed behaviours, however some attempts to widely use other parts of this PR have caused changes. I'll back those out.  But I think this is OK as is.

Reply:
It is also a pattern used by VirtualThreadPool and TryExecutor, both of which add methods to their base-type, but still implement their base-type.